lib/inertia_rails/controller.rb

```diff
@@ -77,19 +79,27 @@ def redirect_back(fallback_location:, allow_other_host: true, **options)
       )
     end
 
+    private
+
     def inertia_view_assigns
       return {} unless @_inertia_instance_props
       view_assigns.except(*@_inertia_skip_props)
     end
 
-    private
+    def inertia_configuration
+      self.class._inertia_configuration.bind_controller(self)
+    end
 
-    def inertia_layout
-      layout = ::InertiaRails.layout
+    def inertia_shared_data
+      initial_data = session[:inertia_errors].present? ? {errors: session[:inertia_errors]} : {}
```

Comment:
I used to set `inertia_share(errors: {})` to mimic Laravel plugin's default, with this change — I always overwrite inertia's errors with my empty hash.

Should we move this to the renderer and apply at the end?

Reply:
I think what you are describing is only working by accident. Before [this change](https://github.com/inertiajs/inertia-rails/pull/111/files#diff-d3e9c2c338687f7af19e477ad05c7dfae9ee7e5ae2fb87adb72f9694b143bc87L11), I believe `inertia_share(errors: {})` would have overwritten the errors from the session (exactly as you described). That change moved the error sharing into a block, which gets evaluated _after_ the plain data is computed, allowing your technique to work. I'd call that unintended behavior because presumably this equivalent looking code just wouldn't work:

```ruby
# This will always overwrite the session errors because this block will be evaluated after the error sharing block from the library
inertia_share do
  {errors: {}}
end
```

IMO, if the application code explicitly sets a prop it _should_ overwrite whatever came out of the library internals, so this PR is actually fixing that.

That said, maybe our default should match the Laravel default and always return something in an errors prop? That seems to be the real thing at issue here. I don't have a strong opinion, but I'd lean towards keeping the default we have since it'd be a potentially breaking change.

I consider that separate from the scope of this PR and won't hold up merging it for this.

(Please let me know if I'm misunderstanding your use case)

Reply:
Oh, great, I just noticed a breaking change, although it appears to be a fix 😂.

Regarding `errors: {}` being the default—that's a significant breaking change, and I would suggest adding a compatibility configuration flag for opting into backward compatibility until the next major release. I also think we should first discuss the possibility of adding this behavior to the protocol. But anyway, that's a separate issue 👍